runtime/web_worker.rs

```diff
@@ -36,38 +37,83 @@ use std::task::Context;
 use std::task::Poll;
 use tokio::sync::Mutex as AsyncMutex;
 
+pub type WorkerId = u32;
+pub type WorkerMessage = Box<[u8]>;
+
 /// Events that are sent to host from child
 /// worker.
 pub enum WorkerEvent {
-  Message(Box<[u8]>),
+  Message(WorkerMessage),
   Error(AnyError),
   TerminalError(AnyError),
 }
 
-pub struct WorkerChannelsInternal {
-  pub sender: mpsc::Sender<WorkerEvent>,
-  pub receiver: mpsc::Receiver<Box<[u8]>>,
+// Channels used for communication with worker's parent
+#[derive(Clone)]
+pub struct WebWorkerInternalHandle {
+  sender: mpsc::Sender<WorkerEvent>,
+  receiver: Rc<RefCell<mpsc::Receiver<WorkerMessage>>>,
+  terminated: Arc<AtomicBool>,
+  isolate_handle: v8::IsolateHandle,
+}
+
+impl WebWorkerInternalHandle {
+  /// Post WorkerEvent to parent as a worker
+  pub fn post_event(&self, event: WorkerEvent) -> Result<(), AnyError> {
+    let mut sender = self.sender.clone();
+    // If the channel is closed,
+    // the worker must have terminated but the termination message has not yet been recieved.
```

Comment:
```suggestion
    // the worker must have terminated but the termination message has not yet been received.
```
(Also: long line. I guess I can't complain if the linter doesn't but consider wrapping it at 80 columns.)